core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainRequest.java

```diff
@@ -139,40 +148,14 @@ public String toString() {
     }
 
     public static ClusterAllocationExplainRequest parse(XContentParser parser) throws IOException {
-        String currentFieldName = null;
-        String index = null;
-        Integer shard = null;
-        Boolean primary = null;
-        XContentParser.Token token;
-        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
-            if (token == XContentParser.Token.FIELD_NAME) {
-                currentFieldName = parser.currentName();
-            } else if (token.isValue()) {
-                if ("index".equals(currentFieldName)) {
-                    index = parser.text();
-                } else if ("shard".equals(currentFieldName)) {
-                    shard = parser.intValue();
-                } else if ("primary".equals(currentFieldName)) {
-                    primary = parser.booleanValue();
-                } else {
-                    throw new ElasticsearchParseException("unexpected field [" + currentFieldName + "] in allocation explain request");
-                }
-
-            } else if (token == XContentParser.Token.START_OBJECT) {
-                // the object was started
-                continue;
-            } else {
-                throw new ElasticsearchParseException("unexpected token [" + token + "] in allocation explain request");
-            }
-        }
+        ClusterAllocationExplainRequest req = new ClusterAllocationExplainRequest();
```

Comment:
I can't! Otherwise the empty constructor would have to be public, and I don't want to make it public